packages/clerk-js/src/core/events.ts

```diff
@@ -1,20 +1,25 @@
 import { createEventBus } from '@clerk/shared/eventBus';
 import type { TokenResource } from '@clerk/types';
 
+import type { SignIn } from './resources';
+
 export const events = {
   TokenUpdate: 'token:update',
   UserSignOut: 'user:signOut',
   EnvironmentUpdate: 'environment:update',
   SessionTokenResolved: 'session:tokenResolved',
+  SignInUpdate: 'signin:update',
 } as const;
 
 type TokenUpdatePayload = { token: TokenResource | null };
+export type SignInUpdatePayload = { resource: SignIn };
```

Comment:
This can be made generic as I expect that its going to be the same for all resources